wave_gtsam/src/preint_imu_factor.cpp

```diff
@@ -0,0 +1,198 @@
+#include "wave/gtsam/preint_imu_factor.hpp"
+
+namespace wave {
+
+template <>
+gtsam::Vector PreintegratedImuFactor<PoseVelBias>::evaluateError(
+  const PoseVelBias &state_i,
+  const PoseVelBias &state_j,
+  const gtsam::imuBias::ConstantBias &bias_i,
+  const gtsam::imuBias::ConstantBias &bias_j,
+  boost::optional<gtsam::Matrix &> H1,
+  boost::optional<gtsam::Matrix &> H2,
+  boost::optional<gtsam::Matrix &> H3,
+  boost::optional<gtsam::Matrix &> H4) const {
+    // Split up the PoseVelBias combined states into pose and vel.
+    // (ignore gps bias)
+    // Then use code adapted from gtsam::CombinedImuFactor.
+    const auto &pose_i = state_i.pose;
+    const auto &pose_j = state_j.pose;
+
+    // Note we use only linear velocity here
+    const auto &vel_i = state_i.vel.tail<3>();
+    const auto &vel_j = state_j.vel.tail<3>();
+
+    // Calculate error wrt bias evolution model (random walk)
+    gtsam::Matrix6 Hbias_i, Hbias_j;
+    gtsam::Vector6 fbias =
+      gtsam::traits<gtsam::imuBias::ConstantBias>::Between(
+        bias_j, bias_i, H4 ? &Hbias_j : 0, H3 ? &Hbias_i : 0)
```

Comment:
blegh I hate the C++ ternary operator so much. so so much.

Reply:
Is this how `gtsam` had it? Do you mind just throwing in a comment maybe as to what this is doing?